pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,42 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		var networkPatchNeeded bool
+		// IF the VMI has a pod network, Check and patch if IP on VMI is not same as the one on Pod IP
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.NetworkSource.Pod == nil {
+				continue
+			}
+			for i, iface := range vmiCopy.Status.Interfaces {
+				if iface.Name == network.Name {
+					if vmiCopy.Status.Interfaces[i].IP != pod.Status.PodIP {
+						vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+						networkPatchNeeded = true
+
+						for j, ip := range iface.IPs {
+							if ip == iface.IP {
+								vmiCopy.Status.Interfaces[i].IPs[j] = pod.Status.PodIP
+							}
+						}
```

Comment:
Aren't these addresses being reported by the agent?
We will overwrite the actual guest addresses if we will do it here.
Perhaps we should only do it if the agent is not configured?
@phoracek what do you think?

Reply:
To be honest, I don't fully get why we need to handle this in watch/vmi.go and not in handler as in https://github.com/kubevirt/kubevirt/pull/3063/files.

Reply:
@phoracek @vladikr I can be wrong here, but I guess `virt-controller` already has the Pod details, so it might be straightforward to update this from here

Reply:
@vladikr about guest-agent, this PR is doing the change to only update agent address only if it's not masquerade https://github.com/kubevirt/kubevirt/pull/3063/files
Are you suggesting any other change here?